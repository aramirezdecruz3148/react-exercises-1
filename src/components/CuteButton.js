import React, { Component } from 'react';

export default class CuteButton extends Component {
  state = {
    count: 0
  }

  handleClick = () => {
    this.setState({ count: this.state.count + 1 });
  }

  render() { 
    const buttonStyles = {
      marginRight: '10px',
      marginLeft: '10px',
      background: '#EE82EE',
      color: 'white'
    };

    const tallyStyles = {
      color: '#32CD32'
    };

    return ( 
      <>
        <button onClick={this.handleClick} style={buttonStyles}>She is CUTE</button>
        <span style={tallyStyles}>Cute Tally: {this.state.count}</span>
      </>
    );
  }
}
