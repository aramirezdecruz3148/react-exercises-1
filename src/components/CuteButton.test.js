import React from 'react';
import { shallow } from 'enzyme';
import CuteButton from './CuteButton';

describe('CuteButton component', () => {
  it('renders CuteButton', () => {
    const wrapper = shallow(<CuteButton />);
    expect(wrapper).toMatchSnapshot();
  });
});
