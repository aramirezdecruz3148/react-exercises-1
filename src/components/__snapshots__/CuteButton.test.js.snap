// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CuteButton component renders CuteButton 1`] = `
<Fragment>
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "#EE82EE",
        "color": "white",
        "marginLeft": "10px",
        "marginRight": "10px",
      }
    }
  >
    She is CUTE
  </button>
  <span
    style={
      Object {
        "color": "#32CD32",
      }
    }
  >
    Cute Tally: 
    0
  </span>
</Fragment>
`;
